From 4d984701dd6440fdc541fb803e72bd59c1950d4d Mon Sep 17 00:00:00 2001
From: Thales Elero Cervi <thaleselero.cervi@windriver.com>
Date: Wed, 18 Sep 2024 19:01:29 -0300
Subject: [PATCH] qemuMonitorCreateObjectPropsWrap: Open-code in
 qemuBuildMemoryBackendProps

There's just one caller left. Since qemuBuildMemoryBackendProps is too
complex to be modified for now, just move the adding of 'id' and 'qom'
type directly into the function.

Signed-off-by: Peter Krempa <pkrempa@redhat.com>
Reviewed-by: Michal Privoznik <mprivozn@redhat.com>
[ Patch defuzzed for libvirt 7.0.0]
Signed-off-by: Thales Elero Cervi <thaleselero.cervi@windriver.com>
---
 src/qemu/qemu_command.c |  6 ++++--
 src/qemu/qemu_monitor.c | 14 --------------
 src/qemu/qemu_monitor.h |  4 ----
 3 files changed, 4 insertions(+), 20 deletions(-)

diff --git a/src/qemu/qemu_command.c b/src/qemu/qemu_command.c
index fa40f36cf25..f3462c528ce 100644
--- a/src/qemu/qemu_command.c
+++ b/src/qemu/qemu_command.c
@@ -3256,10 +3256,12 @@ qemuBuildMemoryBackendProps(virJSONValuePtr *backendProps,
         rc = 0;
     }
 
-    if (!(*backendProps = qemuMonitorCreateObjectPropsWrap(backendType, alias,
-                                                           &props)))
+    if (virJSONValueObjectPrependString(props, "id", alias) < 0 ||
+        virJSONValueObjectPrependString(props, "qom-type", backendType) < 0)
         return -1;
 
+    *backendProps = g_steal_pointer(&props);
+
     return rc;
 }
 
diff --git a/src/qemu/qemu_monitor.c b/src/qemu/qemu_monitor.c
index 2e280b82689..4665f451640 100644
--- a/src/qemu/qemu_monitor.c
+++ b/src/qemu/qemu_monitor.c
@@ -3029,20 +3029,6 @@ qemuMonitorAddDeviceArgs(qemuMonitorPtr mon,
 }
 
 
-virJSONValuePtr
-qemuMonitorCreateObjectPropsWrap(const char *type,
-                                 const char *alias,
-                                 virJSONValuePtr *props)
-{
-    if (virJSONValueObjectPrependString(*props, "id", alias) < 0 ||
-        virJSONValueObjectPrependString(*props, "qom-type", type))
-        return NULL;
-
-    return g_steal_pointer(props);
-}
-
-
-
 /**
  * qemuMonitorCreateObjectProps:
  * @propsret: returns full object properties
diff --git a/src/qemu/qemu_monitor.h b/src/qemu/qemu_monitor.h
index a46b971a33f..792b74a67ea 100644
--- a/src/qemu/qemu_monitor.h
+++ b/src/qemu/qemu_monitor.h
@@ -1006,10 +1006,6 @@ int qemuMonitorAddDeviceWithFd(qemuMonitorPtr mon,
 int qemuMonitorDelDevice(qemuMonitorPtr mon,
                          const char *devalias);
 
-virJSONValuePtr qemuMonitorCreateObjectPropsWrap(const char *type,
-                                                 const char *alias,
-                                                 virJSONValuePtr *props);
-
 int qemuMonitorCreateObjectProps(virJSONValuePtr *propsret,
                                  const char *type,
                                  const char *alias,
-- 
2.34.1

